### PR TITLE
Remove CreateIndexRequest.addMapping(type, string, xcontenttype)

### PR DESCRIPTION
**Pull request by romseygeek**
We still have a number of places, mainly in test code but some in production, that
are building mappings with a named type as the root of a map.  CreateIndexRequest
handles this automatically, but PutMappingRequest does not, which is a bit trappy -
we can get situations like #50359 where the same mapping will work when an
index is created but fail on an update.

This commit is a first step to removing the leniency in CreateIndexRequest so that
we can catch mappings with a named type root earlier.

Relates to #41059

### CODE REVIEW

**Comment by romseygeek** [comment]
@elasticmachine update branch

File: server/src/test/java/org/elasticsearch/gateway/RecoveryFromGatewayIT.java [hunk 1]
```diff
@@ -98,10 +97,10 @@ public void testOneNodeRecoverFromGateway() throws Exception {
 
         internalCluster().startNode();
 
-        String mapping = Strings.toString(XContentFactory.jsonBuilder().startObject().startObject("type1")
-            .startObject("properties").startObject("appAccountIds").field("type", "text").endObject().endObject()
+        String mapping = Strings.toString(XContentFactory.jsonBuilder().startObject()
+            .startObject("properties").startObject("appAccountIds").field("type", "text").endObject()
```
Comment (nik9000) [inline server/src/test/java/org/elasticsearch/gateway/RecoveryFromGatewayIT.java]: I try to re-indent these when I see them to make them a bit more readable.

**Review by nik9000 — APPROVED** [review]
LGTM

**Comment by romseygeek** [comment]
@elasticmachine update branch

**Pull request merged**